yb-voyager/src/migassessment/sizing.go

```diff
@@ -115,416 +172,507 @@ func SizingAssessment() error {
+				ShardedTables:                   []SourceDBMetadata{},
+				VCPUsPerInstance:                previousRecommendation.VCPUsPerInstance,
+				MemoryPerCore:                   previousRecommendation.MemoryPerCore,
+				NumNodes:                        previousRecommendation.NumNodes,
+				OptimalSelectConnectionsPerNode: previousRecommendation.OptimalSelectConnectionsPerNode,
+				OptimalInsertConnectionsPerNode: previousRecommendation.OptimalInsertConnectionsPerNode,
+				ParallelVoyagerJobs:             previousRecommendation.ParallelVoyagerJobs,
+				ColocatedSize:                   0,
+				ShardedSize:                     0,
+				EstimatedTimeInMinForImport:     previousRecommendation.EstimatedTimeInMinForImport,
+				FailureReasoning:                failureReasoning,
+			}
+		}
+	}
+	// Return updated recommendation map
+	return recommendation
+}
 
-	for rows.Next() {
-		var r1 ExpDataColocatedLimit
-		var cumulativeObjectCount int64 = 0
+/*
+shardingBasedOnOperations performs sharding based on operations (reads and writes) per second, taking into account colocated limits.
+It updates the existing recommendations with information about colocated and sharded tables based on operations.
+Parameters:
+  - sourceIndexMetadata: A slice of SourceDBMetadata structs representing source indexes.
+  - colocatedLimits: A slice of ExpDataColocatedLimit structs representing colocated limits.
+  - recommendation: A map where the key is the number of vCPUs per instance and the value is an IntermediateRecommendation struct.
+
+Returns:
+  - An updated map of recommendations where sharding information based on operations has been incorporated.
+*/
+func shardingBasedOnOperations(sourceIndexMetadata []SourceDBMetadata,
+	colocatedLimits []ExpDataColocatedLimit, recommendation map[int]IntermediateRecommendation) map[int]IntermediateRecommendation {
+
+	for _, colocatedLimit := range colocatedLimits {
+		var colocatedObjects []SourceDBMetadata
+		var cumulativeColocatedSizeSum float64 = 0
+		var numColocated int = 0
 		var cumulativeSelectOpsPerSec int64 = 0
 		var cumulativeInsertOpsPerSec int64 = 0
-		var cumulativeSizeSum float64 = 0
-		var colocatedObjects []SourceDBMetadata
-		var currentReasoning string
-		var allObjectsColocated = true
 
-		if err := rows.Scan(&r1.maxColocatedSizeSupported, &r1.numCores, &r1.memPerCore, &r1.maxSupportedNumTables,
-			&r1.minSupportedNumTables, &r1.maxSupportedSelectsPerCore, &r1.maxSupportedInsertsPerCore); err != nil {
-			return nil, 0.0, ExpDataColocatedLimit{}, nil, previousReasoning, fmt.Errorf("cannot fetch data from experiment data table with query [%s]: %w", query, err)
-		}
-		currentReasoning = fmt.Sprintf("Recommended instance with %vvCPU and %vGiB memory could ",
-			r1.numCores.Float64, r1.numCores.Float64*r1.memPerCore.Float64)
+		// Get previous recommendation for the current num of cores
+		previousRecommendation := recommendation[int(colocatedLimit.numCores.Float64)]
```

Comment:
Ques: why are we storing `numCores` as `Float` not as `int` inside the ExpDB?

Reply:
No particular reason. Although, the type conversion will still be required if we do ```sql.NullInt64``` instead of current ```sql.NullFloat64``` as suggested by you . It will not help optimise the code. 
CC: @hbhanawat what do you think?

Reply:
yeah, its just a minor suggestion. Since it will avoid float --> int conversion in your code.